### PR TITLE
Feature/1151 update osmosis deps

### LINKED ISSUE #1151: Bump osmosis-types to v22
We are currently using a version of osmosis-types from a previous version of of Osmosis - we need to bump this to the latest (v22) in order to add CL pools.

### PR DESCRIPTION
**Pull request by joe-bowman**
## 1. Summary
Fixes #1151 - upgrade osmosis types to v22.0.5, in preparation for adding CL pools types to claims.

## 2.Type of change

- [ ] Update dependencies

## 3. Implementation details

- Update types to v22.0.5 - `gamm`, `lockup`, `osmomath`, `osmoutils`.
- Add `concentrated-liquidity` and `poolmanager` modules.
- Remove references to `tx/amino` in pb.go files
- Update gogoproto to use tm0.34 compatible gotoproto, else types fail to register.

## Summary by CodeRabbit

- **New Features**
	- Introduced concentrated liquidity management for third-party chain integration, including creating, managing, and withdrawing from liquidity positions.
	- Added governance proposals for concentrated liquidity pools, including creation and tick spacing adjustments.
	- Implemented concentrated liquidity events and incentives tracking.
	- Enhanced mathematical functions for liquidity and price calculations in concentrated liquidity contexts.
	- Updated GAMM module with new pool types, error handling, and event management.
	- Improved pool management with new methods for swaps, liquidity adjustments, and governance controls across various pool models.

- **Refactor**
	- Consolidated interface definitions and updated method signatures across the Osmosis protocol to enhance clarity and consistency.
	- Refined error handling and message formatting in GAMM and concentrated liquidity modules.
	- Optimized key generation functions for efficient data storage and retrieval in concentrated liquidity pools.

- **Documentation**
	- Added comprehensive documentation on key space format and expectations in the concentrated liquidity module.

- **Bug Fixes**
	- Corrected error types and messages to ensure accurate error reporting and handling in the GAMM module.

### CODE REVIEW

File: third-party-chains/osmosis-types/concentrated-liquidity/model/pool.go [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+		return osmomath.BigDec{}, fmt.Errorf("base asset denom (%s) is not in pool with (%s, %s) pair", baseAssetDenom, p.Token0, p.Token1)
+	}
+	// validate quote asset is in pool
+	if quoteAssetDenom != p.Token0 && quoteAssetDenom != p.Token1 {
+		return osmomath.BigDec{}, fmt.Errorf("quote asset denom (%s) is not in pool with (%s, %s) pair", quoteAssetDenom, p.Token0, p.Token1)
+	}
+
+	// The reason why we convert the result to Dec and then back to BigDec is to temporarily
+	// maintain backwards compatibility with the original implementation.
+	// TODO: remove before https://github.com/osmosis-labs/osmosis/issues/5726 is complete
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/concentrated-liquidity/model/pool.go:123]: @coderabbitai - files within the third-party-chains/osmosis-types are vendored from an upstream repository and as such, TODOs can be ignored safely.

[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/concentrated-liquidity/model/pool.go:123]: Wasn't coderabbit :D

File: third-party-chains/osmosis-types/gamm/pool-models/balancer/amm.go [hunk 2]
```diff
@@ -3,12 +3,11 @@ package balancer
...
 
-	sdkioerrors "cosmossdk.io/errors"
-	sdkmath "cosmossdk.io/math"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	"github.com/quicksilver-zone/quicksilver/third-party-chains/osmosis-types/gamm"
+	errorsmod "cosmossdk.io/errors"
+
+	types "github.com/quicksilver-zone/quicksilver/third-party-chains/osmosis-types/gamm"
```
[the remark this answers is not in the document]
Comment (faddat) [reply third-party-chains/osmosis-types/gamm/pool-models/balancer/amm.go:10]: nope, this is the standard, and doing it the way you describe would be inconsistent.

**Comment by faddat** [comment]
it is doing the thing again.  I cannot figure out why.

@coderabbitai why is only amd64/linux failing?

**Comment by faddat** [comment]
@joe-bowman I made #1153 in order to resolve the platform specific issues.

File: third-party-chains/osmosis-types/gamm/pool-models/balancer/pool.go [hunk 3]
```diff
@@ -371,7 +379,7 @@ func (p *Pool) updateAllWeights(newWeights []PoolAsset) {
 	if len(p.PoolAssets) != len(newWeights) {
 		panic("updateAllWeights called with invalid input, len(newWeights) != len(existingWeights)")
 	}
-	totalWeight := sdk.ZeroInt()
+	totalWeight := osmomath.ZeroInt()
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/gamm/pool-models/balancer/pool.go:382]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/gamm/pool-models/balancer/msgs.go [hunk 4]
```diff
@@ -32,12 +32,12 @@ func NewMsgCreateBalancerPool(
...
 
-func (MsgCreateBalancerPool) Route() string { return gamm.RouterKey }
-func (MsgCreateBalancerPool) Type() string  { return TypeMsgCreateBalancerPool }
+func (msg MsgCreateBalancerPool) Route() string { return types.RouterKey }
+func (msg MsgCreateBalancerPool) Type() string  { return TypeMsgCreateBalancerPool }
 func (msg MsgCreateBalancerPool) ValidateBasic() error {
 	_, err := sdk.AccAddressFromBech32(msg.Sender)
 	if err != nil {
-		return sdkioerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid sender address (%s)", err)
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid sender address (%s)", err)
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/gamm/pool-models/balancer/msgs.go:40]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/gamm/pool-models/balancer/pool_params.go [hunk 5]
```diff
@@ -69,10 +68,10 @@
 	return nil
 }
 
-func (params PoolParams) GetPoolSwapFee() sdk.Dec {
+func (params PoolParams) GetPoolSpreadFactor() osmomath.Dec {
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/gamm/pool-models/balancer/pool_params.go:71]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/gamm/pool-models/balancer/pool_params.go [hunk 6]
```diff
@@ -3,41 +3,40 @@ package balancer
...
-	"github.com/quicksilver-zone/quicksilver/third-party-chains/osmosis-types/gamm"
+	types "github.com/quicksilver-zone/quicksilver/third-party-chains/osmosis-types/gamm"
+	"github.com/quicksilver-zone/quicksilver/third-party-chains/osmosis-types/osmomath"
 )
 
-func NewPoolParams(swapFee, exitFee sdk.Dec, params *SmoothWeightChangeParams) PoolParams {
+func NewPoolParams(spreadFactor, exitFee osmomath.Dec, params *SmoothWeightChangeParams) PoolParams {
 	return PoolParams{
-		SwapFee:                  swapFee,
+		SwapFee:                  spreadFactor,
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/gamm/pool-models/balancer/pool_params.go:12]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/gamm/pool-models/balancer/amm.go [hunk 7]
```diff
@@ -72,13 +71,13 @@
...
-func ValidateUserSpecifiedWeight(weight sdkmath.Int) error {
+func ValidateUserSpecifiedWeight(weight osmomath.Int) error {
 	if !weight.IsPositive() {
-		return sdkioerrors.Wrap(gamm.ErrNotPositiveWeight, weight.String())
+		return errorsmod.Wrap(types.ErrNotPositiveWeight, weight.String())
 	}
 
 	if weight.GTE(MaxUserSpecifiedWeight) {
-		return sdkioerrors.Wrap(gamm.ErrWeightTooLarge, weight.String())
+		return errorsmod.Wrap(types.ErrWeightTooLarge, weight.String())
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/gamm/pool-models/balancer/amm.go:80]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/gamm/pool-models/balancer/amm.go [hunk 8]
```diff
@@ -57,7 +56,7 @@
 }
 
 // assumes 0 < d < 1
-func poolAssetsMulDec(base []PoolAsset, d sdk.Dec) []PoolAsset {
+func poolAssetsMulDec(base []PoolAsset, d osmomath.Dec) []PoolAsset {
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/gamm/pool-models/balancer/amm.go:59]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/concentrated-liquidity/errors.go [hunk 9]
```diff
@@ -0,0 +1,932 @@
...
+}
+
+type InvalidActionPrefixError struct {
+	ActionPrefix string
+	ValidActions []string
+}
+
+func (e InvalidActionPrefixError) Error() string {
+	return fmt.Sprintf("invalid action prefix (%s). Valid actions: %s", e.ActionPrefix, e.ValidActions)
+}
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/concentrated-liquidity/errors.go:932]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/concentrated-liquidity/errors.go [hunk 10]
```diff
@@ -0,0 +1,932 @@
...
+var (
+	ErrKeyNotFound                        = errors.New("key not found")
+	ErrValueParse                         = errors.New("value parse error")
+	ErrPositionNotFound                   = errors.New("position not found")
+	ErrZeroPositionId                     = errors.New("invalid position id, cannot be 0")
+	ErrPermissionlessPoolCreationDisabled = errors.New("permissionless pool creation is disabled for the concentrated liquidity module")
+	ErrZeroLiquidity                      = errors.New("liquidity cannot be 0")
+	ErrNextTickInfoNil                    = errors.New("next tick info cannot be nil")
+	ErrPoolNil                            = errors.New("pool cannot be nil")
+)
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/concentrated-liquidity/errors.go:22]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/gamm/pool-models/balancer/amm.go [hunk 11]
```diff
@@ -186,9 +185,9 @@
 }
 
 // feeRatio returns the fee ratio that is defined as follows:
-// 1 - ((1 - normalizedTokenWeightOut) * swapFee)
-func feeRatio(normalizedWeight, swapFee sdk.Dec) sdk.Dec {
-	return sdk.OneDec().Sub((sdk.OneDec().Sub(normalizedWeight)).Mul(swapFee))
+// 1 - ((1 - normalizedTokenWeightOut) * spreadFactor)
+func feeRatio(normalizedWeight, spreadFactor osmomath.Dec) osmomath.Dec {
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/gamm/pool-models/balancer/amm.go:189]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/gamm/pool-models/balancer/amm.go [hunk 12]
```diff
@@ -236,7 +235,7 @@
 	for _, coin := range tokensIn {
 		_, ok := poolAssetsByDenom[coin.Denom]
 		if !ok {
-			return sdkioerrors.Wrapf(gamm.ErrDenomNotFoundInPool, invalidInputDenomsErrFormat, coin.Denom)
+			return errorsmod.Wrapf(types.ErrDenomNotFoundInPool, invalidInputDenomsErrFormat, coin.Denom)
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/gamm/pool-models/balancer/amm.go:238]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/gamm/pool-models/stableswap/pool.go [hunk 13]
```diff
@@ -4,47 +4,70 @@
...
 
 	return pool, nil
 }
 
 func (p Pool) GetAddress() sdk.AccAddress {
-	addr, err := addressutils.AccAddressFromBech32(p.Address, "")
+	addr, err := sdk.AccAddressFromBech32(p.Address)
 	if err != nil {
 		panic(fmt.Sprintf("could not bech32 decode address of pool with id: %d", p.GetId()))
 	}
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/gamm/pool-models/stableswap/pool.go:73]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/gamm/pool-models/stableswap/pool.go [hunk 14]
```diff
@@ -63,15 +86,15 @@
...
 	return p.PoolParams.SwapFee
 }
 
-func (p Pool) GetExitFee(ctx sdk.Context) sdk.Dec {
+func (p Pool) GetExitFee(ctx sdk.Context) osmomath.Dec {
 	return p.PoolParams.ExitFee
 }
 
-func (Pool) IsActive(ctx sdk.Context) bool {
+func (p Pool) IsActive(ctx sdk.Context) bool {
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/gamm/pool-models/stableswap/pool.go:97]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

File: third-party-chains/osmosis-types/concentrated-liquidity/model/pool.go [hunk 15]
```diff
@@ -0,0 +1,326 @@
...
+		CurrentSqrtPrice:     osmomath.ZeroBigDec(),
+		CurrentTick:          0,
+		CurrentTickLiquidity: osmomath.ZeroDec(),
+		Token0:               denom0,
+		Token1:               denom1,
+		TickSpacing:          tickSpacing,
+		ExponentAtPriceOne:   types.ExponentAtPriceOne,
+		SpreadFactor:         spreadFactor,
+	}
+	return pool, nil
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply third-party-chains/osmosis-types/concentrated-liquidity/model/pool.go:55]: @coderabbitai - all files within third-party-chains are vendored files from another project, included to provide types for cross chain queries. We want to keep them as close to the original as possible.

**Pull request merged**